core/src/main/scala/cats/data/Validated.scala

```diff
@@ -1166,6 +1166,11 @@ private[data] trait ValidatedFunctions {
    */
   final def condNel[E, A](test: Boolean, a: => A, e: => E): ValidatedNel[E, A] =
     if (test) validNel(a) else invalidNel(e)
+
+  /**
+   * Cached value of `Valid(())` to avoid allocations for a common case.
+   */
+  final def unit[E]: Validated[E, Unit] = Validated.validUnit
```

Comment:
Thanks! Do we need the type parameter to help with inference or would `Validated[Nothing, Unit]` work fine?

Reply:
Not really sure, I followed an example from `Either.unit` for consistency.

Reply:
In some obscure cases, it might help, actually, but I don't know off the top of my head

Reply:
I see, I didn't find any clues in https://github.com/typelevel/cats/pull/3346. `Either.unit` works well enough so this seems fine then :)

Reply:
Yeah, hard to say. I also wondered if its an artifact of old compilers e.g. 2.11 that couldn't handle these as well.

Reply:
Possibly yes, I remember stumbling upon such cases a few years back with `zio-saga` library, but not sure any more if this behaviour still exists in modern versions.